src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/SasTokenGeneratorService.java

```diff
@@ -13,55 +12,46 @@
 import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.ServiceConfigNotFoundException;
 import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.UnableToGenerateSasTokenException;
 
-import java.net.URISyntaxException;
-import java.security.InvalidKeyException;
-import java.time.ZoneOffset;
-import java.util.Date;
-import java.util.EnumSet;
-
-import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.LIST;
-import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.WRITE;
-import static java.time.LocalDateTime.now;
+import java.time.OffsetDateTime;
 
 @EnableConfigurationProperties(AccessTokenProperties.class)
 @Service
 public class SasTokenGeneratorService {
 
     private static final Logger log = LoggerFactory.getLogger(SasTokenGeneratorService.class);
 
-    private final CloudBlobClient cloudBlobClient;
+    private final BlobServiceClient blobServiceClient;
     private final AccessTokenProperties accessTokenProperties;
+    private static final String PERMISSION_WRITE_LIST = "wl";
 
     public SasTokenGeneratorService(
-        CloudBlobClient cloudBlobClient,
+        BlobServiceClient blobServiceClient,
         AccessTokenProperties accessTokenProperties
     ) {
-        this.cloudBlobClient = cloudBlobClient;
+        this.blobServiceClient = blobServiceClient;
         this.accessTokenProperties = accessTokenProperties;
     }
 
     public String generateSasToken(String serviceName) {
-        StorageUri storageAccountUri = cloudBlobClient.getStorageUri();
+        String storageAccountUri = blobServiceClient.getAccountUrl();
         log.info("SAS Token request received for service {}. Account URI: {}", serviceName, storageAccountUri);
+        TokenConfig config = getTokenConfigForService(serviceName);
 
         try {
-            return cloudBlobClient
-                .getContainerReference(serviceName)
-                .generateSharedAccessSignature(createSharedAccessPolicy(serviceName), null);
-
-        } catch (URISyntaxException | StorageException | InvalidKeyException e) {
+            return blobServiceClient
+                .getBlobContainerClient(serviceName)
+                .generateSas(createSharedAccessPolicy(config));
+        } catch (Exception e) {
             throw new UnableToGenerateSasTokenException(e);
         }
     }
 
-    private SharedAccessBlobPolicy createSharedAccessPolicy(String serviceName) {
-        TokenConfig config = getTokenConfigForService(serviceName);
-
-        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
-        policy.setPermissions(EnumSet.of(WRITE, LIST));
-        policy.setSharedAccessExpiryTime(Date.from(now().plusSeconds(config.getValidity()).toInstant(ZoneOffset.UTC)));
```

Comment:
what about the timezone ?

Reply:
OffsetDateTime  is A date-time with an offset from UTC/Greenwich .

Reply:
depends on servers. i guess our pods are in UK South

Reply:
I hope the timezone is set correctly otherwise this would block our consumer of file uploads

Reply:
I will set to UTC explicitly